docs/data/date-pickers/time-range-picker/time-range-picker.md

```diff
@@ -1,18 +1,91 @@
+
+- [Time Range Field](/x/react-date-pickers/time-range-field/)
+- [Digital Clock](/x/react-date-pickers/digital-clock/)
+
+## Uncontrolled vs. controlled value
+
+The value of the component can be uncontrolled or controlled.
+
+{{"demo": "TimeRangePickerValue.js"}}
+
+:::info
+
+- The value is **controlled** when its parent manages it by providing a `value` prop.
+- The value is **uncontrolled** when it is managed by the component's own internal state. This state can be initialized using the `defaultValue` prop.
+
+Learn more about the _Controlled and uncontrolled_ pattern in the [React documentation](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components).
+:::
+
+## Available components
+
+The component is available in three variants:
+
+- The `DesktopTimeRangePicker` component which works best for mouse devices and large screens.
+  It renders the views inside a popover and a field for keyboard editing.
+
+- The `MobileTimeRangePicker` component which works best for touch devices and small screens.
+  It renders the views inside a modal and and a field for keyboard editing.
+
+- The `TimeRangePicker` component which renders `DesktopTimeRangePicker` or `MobileTimeRangePicker` depending on the device it runs on.
+
+{{"demo": "ResponsiveTimeRangePickers.js"}}
+
+By default, the `TimeRangePicker` component renders the desktop version if the media query [`@media (pointer: fine)`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer) matches.
+This can be customized with the `desktopModeMediaQuery` prop.
+
 :::warning
-This component isn't available yet, but it is planned—you can 👍 upvote [this GitHub issue](https://github.com/mui/mui-x/issues/4460) to help us prioritize it.
-Please don't hesitate to leave a comment there to describe your needs, especially if you have a use case we should address or you're facing specific pain points with your current solution.
+Responsive components can suffer some inconsistencies between testing environments if media query is not supported.
+Please refer to [this section](/x/react-date-pickers/base-concepts/#testing-caveats) for solutions.
 :::
 
-The Time Range Picker would let users select a range of time values.
+## Form props
+
+The component supports the `disabled`, `readOnly` and `name` props:
+
+{{"demo": "FormPropsTimeRangePickers.js"}}
+
+:::success
+The `name` prop is not available when using the Time Range Picker with the Multi Input Time Range Field.
+:::
+
+## Customization
+
+### Use a multi input field
+
+You can pass the `MultiInputTimeRangeField` component to the Time Range Picker to use it for keyboard editing:
+
+{{"demo": "MultiInputTimeRangePicker.js"}}
+
+:::info
+You can find more information in a [dedicated documentation page section](/x/react-date-pickers/custom-field/#usage-inside-a-range-picker).
+:::
+
+### Change end time label
+
+{{"demo": "CustomizedBehaviorTimeRangePicker.js"}}
```

Comment:
It's worth adding 1-2 sentence explaining what the demo is doing, it took me some time to understand, I was looking for the field's label

Reply:
Good spot. 👍 
I initially added the demo and later forgot to add any sort of text. 🙈

Reply:
I added a couple of sentences/paragraphs.
Feel free to suggest improvements. 👌